homeassistant/components/switchbee/light.py

```diff
@@ -0,0 +1,175 @@
+
+from .const import DOMAIN
+from .coordinator import SwitchBeeCoordinator
+from .entity import SwitchBeeDeviceEntity
+
+MAX_BRIGHTNESS = 255
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _brightness_hass_to_switchbee(value: int) -> int:
+    """Convert hass brightness to SwitchBee."""
+    sb_brightness = int(100 * (value) / MAX_BRIGHTNESS)
+    # SwitchBee maximum brightness is 99
+    return sb_brightness if sb_brightness != 100 else 99
+
+
+def _brightness_switchbee_to_hass(value: int) -> int:
+    """Convert SwitchBee brightness to hass."""
+    if value == 99:
+        value = 100
+    return round(value * MAX_BRIGHTNESS / 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SwitchBee light."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SwitchBeeLightEntity(switchbee_device, coordinator)
+        for switchbee_device in coordinator.data.values()
+        if switchbee_device.type == DeviceType.Dimmer
+    )
+
+
+class SwitchBeeLightEntity(SwitchBeeDeviceEntity[SwitchBeeDimmer], LightEntity):
+    """Representation of an SwitchBee light."""
+
+    def __init__(
+        self,
+        device: SwitchBeeDimmer,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the SwitchBee light."""
+        super().__init__(device, coordinator)
+        self._attr_is_on = False
+        self._is_online = True
+        self._attr_brightness: int = 0
+        self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        self._attr_available = True
+
+        self._update_attrs_from_coordinator()
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._is_online and super().available
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_attrs_from_coordinator()
+        super()._handle_coordinator_update()
+
+    def _update_attrs_from_coordinator(self) -> None:
+        async def async_refresh_state() -> None:
+            """Refresh the device state in the Central Unit.
+
+            This function addresses issue of a device that came online back but still report
+            unavailable state (-1).
+            Such device (offline device) will keep reporting unavailable state (-1)
+            until it has been actuated by the user (state changed to on/off).
+
+            With this code we keep trying setting dummy state for the device
+            in order for it to start reporting its real state back (assuming it came back online)
+
+            """
+
+            try:
+                await self.coordinator.api.set_state(self._device.id, "dummy")
+            except SwitchBeeDeviceOfflineError:
+                return
+            except SwitchBeeError:
+                return
```

Comment:
I think it should be possible to move this to `SwitchBeeDeviceEntity` so the code isn't duplicated

Reply:
Should we do the refactoring here? or in a separate following PR after having the other platforms in?